jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx

```diff
@@ -30,28 +34,46 @@ export type usePipelineResponse = {
 export const usePipeline = (
   ppsContext: PpsContext | undefined,
   settings: MountSettings,
-  saveNotebookMetaData: (metadata: SameMetadata) => void,
+  saveNotebookMetaData: (metadata: PpsMetadata) => void,
 ): usePipelineResponse => {
   const [loading, setLoading] = useState(false);
-  const [pipelineName, setPipelineName] = useState('');
+  const [pipeline, _setPipeline] = useState({name: ''} as Pipeline);
   const [imageName, setImageName] = useState('');
   const [inputSpec, setInputSpec] = useState('');
   const [requirements, setRequirements] = useState('');
   const [errorMessage, setErrorMessage] = useState('');
   const [responseMessage, setResponseMessage] = useState('');
   const [currentNotebook, setCurrentNotebook] = useState('None');
 
+  const setPipeline = (input: string) => {
+    if (input === '') {
+      _setPipeline({name: ''} as Pipeline);
+      return;
+    }
+    const parts = input.split(/\/(.*)/s, 2);
+    if (parts.length === 1) {
+      _setPipeline({name: input} as Pipeline);
+    } else {
+      _setPipeline({name: parts[1], project: {name: parts[0]}} as Pipeline);
+    }
+  };
+
   useEffect(() => {
     setImageName(
-      ppsContext?.config?.environments.default.image_tag ??
-        settings.defaultPipelineImage,
+      ppsContext?.metadata?.config.image ?? settings.defaultPipelineImage,
+    );
+    _setPipeline(
+      ppsContext?.metadata?.config.pipeline ?? ({name: ''} as Pipeline),
     );
-    setPipelineName(ppsContext?.config?.metadata.name ?? '');
-    setRequirements(ppsContext?.config?.notebook.requirements ?? '');
+    setRequirements(ppsContext?.metadata?.config.requirements ?? '');
     setResponseMessage('');
-    if (ppsContext?.config?.run.input) {
-      const input = JSON.parse(ppsContext.config.run.input); //TODO: Catch errors
-      setInputSpec(YAML.stringify(input));
+    if (ppsContext?.metadata?.config.input_spec) {
+      try {
+        setInputSpec(YAML.stringify(ppsContext.metadata.config.input_spec));
```

Comment:
Does this yaml stringify work with an empty input spec?

Reply:
yes it does